server/common/auth/mw.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const { createError } = require('../errors');
+const HttpStatus = require('http-status');
+const Role = require('../../../common/config/role');
+
+const { UNAUTHORIZED, FORBIDDEN } = HttpStatus;
+
+function authorize(...allowed) {
+  allowed.push(Role.ADMIN);
+  return ({ user }, res, next) => {
+    return user
```

Comment:
Not a good use of ternary. You can write this nicer and more readable with two `if`s and default rule.